webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js

```diff
@@ -0,0 +1,147 @@
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const API_KEY = `get-reports-${hostName}`;
+  const { reports, itemCount } = useSelector(state =>
+    selectAPIResponse(state, API_KEY)
+  );
+  const { perPage: settingsPerPage = 20 } = useForemanSettings() || {};
+  const status = useSelector(state => selectAPIStatus(state, API_KEY));
+  const fetchReports = useCallback(
+    ({ search: searchParam, per_page: perPageParam, page: pageParam } = {}) => {
+      const {
+        page: urlPage,
+        perPage: urlPerPage,
+        search: urlSearch,
+      } = getUrlParams();
+      const search = searchParam !== undefined ? searchParam : urlSearch;
+      const page = pageParam || urlPage;
+      const per_page = perPageParam || urlPerPage;
+      dispatch(
+        get({
+          key: API_KEY,
+          url: '/config_reports',
+          params: {
+            page,
+            per_page,
+            search: getServerQuery(search),
+          },
+        })
+      );
+      updateUrl({ page, per_page, search });
+    },
+    [API_KEY, dispatch, getServerQuery, getUrlParams, updateUrl]
+  );
+
+  useEffect(() => {
+    fetchReports();
+  }, [fetchReports, history.location]);
+
+  const onPaginationChange = ({ page, per_page }) => {
+    const { search } = getUrlParams();
+    updateUrl({ page, per_page, search });
+  };
+
+  const getServerQuery = useCallback(
+    search => {
+      const serverQuery = [`host = ${hostName}`];
+      if (origin) {
+        serverQuery.push(`origin = ${origin}`);
+      }
+      if (search) {
+        serverQuery.push(`(${search})`);
+      }
+      return serverQuery.join(' AND ');
+    },
+    [origin, hostName]
+  );
+
+  const getUrlParams = useCallback(() => {
+    const params = { page: 1, perPage: settingsPerPage, search: '' };
+    const urlSearch = history.location?.search;
+    const urlParams = urlSearch && new URLSearchParams(urlSearch);
+    if (urlParams) {
+      params.search = urlParams.get('search') || params.search;
+      params.page = Number(urlParams.get('page')) || params.page;
+      params.perPage = Number(urlParams.get('per_page')) || params.perPage;
+    }
+    return params;
+  }, [history.location, settingsPerPage]);
+
+  const updateUrl = useCallback(
+    ({ page, per_page, search = '' }) => {
+      const uri = new URI();
+      uri.search({ page, per_page, search });
+      history.push({ search: uri.search() });
+    },
+    [history]
+  );
+
+  return (
+    <Grid id="new_host_details_insights_tab" hasGutter style={{ padding: 24 }}>
+      <GridItem span={5}>
+        <SearchBar
+          data={{
+            ...getControllerSearchProps('/config_reports'),
+            controller: 'config_reports',
+          }}
+          onSearch={search => fetchReports({ search, page: 1 })}
+          initialQuery={getUrlParams().search}
+        />
+      </GridItem>
+      <GridItem span={4}>
+        <Divider isVertical />
```

Comment:
Can there be some kind of documentation button or explanation tooltip to what this page is?

Reply:
as a follow up we will add more buttons and quick filters around